src/util/Recoil_deepFreezeValue.js

```diff
@@ -64,7 +64,7 @@ function deepFreezeValue(value: mixed) {
 
   Object.freeze(value); // Make all properties read-only
   for (const prop in value) {
-    if (value.hasOwnProperty(prop)) {
+    if (value.hasOwnProperty(prop) && !Object.isFrozen(prop)) {
```

Comment:
This breaks `deepFreezeValue()` because it is checking if the property string key is frozen instead of the property itself.  strings are always frozen so it essentially turns off all freezing of properties.  It causes the tests in `Recoil_core-test.js` to fail.  We can easily fix this by looking up the property itself.

This also isn't perfect, since if we encounter a non-recursive, previously frozen, object that contains unfrozen objects, they won't get frozen.  But, probably a small corner case not worth the extra cost of keeping track of visited objects to otherwise avoid infinite recursion.

Reply:
You are right! My bad, It should be `Object.isFrozen(value[prop])` instead.